app/src/main/java/org/dhis2/usescases/general/ActivityGlobalAbstract.java

```diff
@@ -88,6 +92,14 @@ public void setScreenName(String name) {
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
+        ServerComponent serverComponent = ((App) getApplicationContext()).getServerComponent();
+        if (serverComponent != null) {
+            serverComponent.openIdSession().setSessionCallback(this, () -> {
```

Comment:
If we check this in the onCreate() we only check if the session is expired when the user navigates through activites, right?
Maybe could we check it in the on resume?.

Reply:
It is an observable, it just subscribes to it when the activity is created. So it will be listening the whole time.

Reply:
you are right 🤦🏽